src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java

```diff
@@ -26,45 +26,48 @@
 import java.util.Arrays;
 import java.util.Date;
 import java.util.List;
+import java.util.stream.Collectors;
 
 import org.junit.Before;
 import org.junit.Test;
-import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.annotation.Id;
 import org.springframework.data.elasticsearch.annotations.Document;
 import org.springframework.data.elasticsearch.annotations.Field;
 import org.springframework.data.elasticsearch.annotations.FieldType;
-import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
+import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
+import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
 import org.springframework.data.elasticsearch.utils.IndexInitializer;
-import org.springframework.data.repository.PagingAndSortingRepository;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
 /**
+ * base class for query keyword tests. Implemented by subclasses using ElasticsearchClient and ElasticsearchRestClient
+ * based repositories.
+ *
  * @author Artur Konczak
  * @author Christoph Strobl
  * @author Peter-Josef Meisch
  */
-@RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration("classpath:/repository-query-keywords.xml")
-public class QueryKeywordsTests {
+abstract class QueryKeywordsTests {
 
 	@Autowired private ProductRepository repository;
 
-	@Autowired private ElasticsearchTemplate elasticsearchTemplate;
+	@Autowired private ElasticsearchOperations elasticsearchTemplate;
 
 	@Before
 	public void before() {
 
 		IndexInitializer.init(elasticsearchTemplate, Product.class);
 
-		repository.saveAll(
-				Arrays.asList(Product.builder().id("1").name("Sugar").text("Cane sugar").price(1.0f).available(false).build(),
-						Product.builder().id("2").name("Sugar").text("Cane sugar").price(1.2f).available(true).build(),
-						Product.builder().id("3").name("Sugar").text("Beet sugar").price(1.1f).available(true).build(),
-						Product.builder().id("4").name("Salt").text("Rock salt").price(1.9f).available(true).build(),
-						Product.builder().id("5").name("Salt").text("Sea salt").price(2.1f).available(false).build()));
+		repository.saveAll(Arrays.asList(
```

Comment:
Nit: Extract product objects for readability into variables.

Reply:
done